newsroom/search.py

```diff
@@ -550,12 +550,23 @@ def validate_request(self, search):
                 abort(403, gettext("User does not belong to a company."))
             elif not len(search.products):
                 abort(403, gettext("Your company doesn't have any products defined."))
+            elif len(search.products) and self.is_product_assigned_to_company(search):
```

Comment:
it should probably read like `and not is_product_assigned_to_company` instead of returning `True` when it's not assigned, that's confusing